src/backend/src/utils/notifications.utils.ts

```diff
@@ -0,0 +1,69 @@
+import { Team, Task as Prisma_Task, WBS_Element } from '@prisma/client';
+import { UserWithSettings } from './auth.utils';
+import { HttpException } from './errors.utils';
+
+export type UserWithTeams = UserWithSettings & {
+  teamAsHead: Team | null;
+  teamsAsLead: Team[] | null;
+  teamsAsMember: Team[] | null;
+};
+
+export type TaskWithAssignees = Prisma_Task & {
+  assignees: UserWithSettings[] | null;
+  wbsElement: WBS_Element | null;
+};
+
+export const usersToSlackPings = (users: UserWithSettings[]) => {
+  // https://api.slack.com/reference/surfaces/formatting#mentioning-users
+  return users.map((user) => `<@${user.userSettings?.slackId}>`).join(' ');
+};
+
+/**
+ * Gets the team of a task's assignees.
+ * Assumes all assigness share a team
+ * @param users the users of the task
+ * @returns the team assigned to the task
+ */
+export const getTeamFromTaskAssignees = (users: UserWithTeams[]): Team => {
+  const allTeams = users.map((user) => {
```

Comment:
this is an unclear variable name, usersTeams would be better since it's an array of arrays of teams